### PR TITLE
Arn 76 calculate needs

### PR DESCRIPTION
**Pull request by wood-n**
Current calculated needs are ACCOMMODATION and EDUCATION_TRAINING_AND_EMPLOYABILITY.

The yes/no questions for the following risk questions are added as boolean flags for each need.
Risk of Harm
Risk of Reoffending
Low Scoring Need

A high scoring need is true when the sum of the values for a set of questions for each need is over its threshold.
e.g. Accommodation need 	Sum of Qs (no_fixed_abode, accom_suitability, accom_permanence, accom_location) 	criminogenic need if >= 2

NeedStatus is calculated for each need as 
NEED_IDENTIFIED (where a boolean risk is flagged as true), 
NO_NEED_IDENTIFIED  (where no need is identified and all questions are answered),
or INSUFFICIENT_DATA (where no need is identified and at least one question has not been answered)

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/assessments/api/AssessmentAnswersDto.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package uk.gov.justice.digital.assessments.api
+
+import java.util.*
+
+data class AssessmentAnswersDto (
+
+        val assessmentUuid: UUID,
+        //Question Code -> List Of Answer Codes
+        val answers: Map<String, Set<String>>
+        )
```
Comment (jezhiggins) [inline src/main/kotlin/uk/gov/justice/digital/assessments/api/AssessmentAnswersDto.kt:10]: I wonder if this should look at little more like the shape AssessmentEpisodeDto? and the AnswerDtos it contains. Perhaps even use the AnswerDto?  

Not important at this stage though.

File: src/main/kotlin/uk/gov/justice/digital/assessments/services/AssessmentService.kt [hunk 2]
```diff
@@ -56,6 +57,10 @@ class AssessmentService(private val assessmentRepository: AssessmentRepository)
                 ?: throw EntityNotFoundException("No current Episode for $assessmentUuid")
     }
 
+    fun getCurrentAssessmentAnswers(assessmentUuid: UUID) : AssessmentAnswersDto {
+        return AssessmentAnswersDto(UUID.randomUUID(), emptyMap())
+    }
+
```
Comment (jezhiggins) [inline src/main/kotlin/uk/gov/justice/digital/assessments/services/AssessmentService.kt:63]: I thoroughly approve of this stub :+1:

File: src/main/kotlin/uk/gov/justice/digital/needs/HmppsNeedsApiApplication.kt [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+import org.springframework.boot.autoconfigure.SpringBootApplication
+import org.springframework.boot.runApplication
+
+@SpringBootApplication
+class HmppsNeedsApiApplication
+
+fun main(args: Array<String>) {
+    runApplication<HmppsNeedsApiApplication>(*args)
+}
```
Comment (jezhiggins) [inline src/main/kotlin/uk/gov/justice/digital/needs/HmppsNeedsApiApplication.kt:11]: Do we need this at all? Isn't the existing main ok?

I'm a bit frightened that having two classes annotated with @SpringBootApplication is going to behave is a spooky way.

File: src/main/kotlin/uk/gov/justice/digital/needs/api/AnswerDto.kt [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+import io.swagger.v3.oas.annotations.media.Schema
+
+data class AnswerDto (
+
+        @Schema(description = "Answer ID and value", example = "<RSR01, yes>")
+        val answers: Map<String, String> = emptyMap()
+
+
+//add date that answer was last updated and episode id for answer?
+)
```
Comment (jezhiggins) [inline src/main/kotlin/uk/gov/justice/digital/needs/api/AnswerDto.kt:12]: Can we use the api.AnswerDto?

**Review by jezhiggins — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/needs/HmppsNeedsApiApplication.kt (hunk 3 above)
Comment (jezhiggins) [reply src/main/kotlin/uk/gov/justice/digital/needs/HmppsNeedsApiApplication.kt:11]: Same with the jpa and OpenApi config.

Comment (wood-n) [reply src/main/kotlin/uk/gov/justice/digital/needs/HmppsNeedsApiApplication.kt]: Taken out the duplicate files :)

**Review by jezhiggins — APPROVED** [review]

**Comment by jezhiggins** [comment]
Could you add a brief description up at the top, outlining how the needs are defined and scored?

**Pull request merged**